include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -324,7 +400,31 @@ SpatialOperatorBase<dim, Number>::initialize_operators(std::string const & dof_i
   MassOperatorData<dim> mass_operator_data;
   mass_operator_data.dof_index  = get_dof_index_velocity();
   mass_operator_data.quad_index = get_quad_index_velocity_linear();
-  mass_operator.initialize(*matrix_free, constraint_dummy, mass_operator_data);
+  mass_operator.initialize(*matrix_free, constraint_u, mass_operator_data);
+
+  // Mass solver for HDIV (instead of inverse mass operator as it is unavailiable for hdiv)
+  if(param.spatial_discretization == SpatialDiscretization::HDIV)
+  {
+    Krylov::SolverDataCG solver_data;
+    solver_data.max_iter             = this->param.solver_data_mass.max_iter;
+    solver_data.solver_tolerance_abs = this->param.solver_data_mass.abs_tol;
+    solver_data.solver_tolerance_rel = this->param.solver_data_mass.rel_tol;
+
+    if(param.preconditioner_mass == PreconditionerMass::PointJacobi)
+    {
+      mass_preconditioner =
+        std::make_shared<JacobiPreconditioner<MassOperator<dim, dim, Number>>>(this->mass_operator);
+      solver_data.use_preconditioner = true;
+    }
+    else if(param.preconditioner_mass == PreconditionerMass::None)
+    {
+      solver_data.use_preconditioner = false;
+    }
```

Comment:
I think we normally start with None, then the different preconditioners that are implemented, and finally an else branch with AssertThrow(false).

Imagine someone adds `Multigrid` as option for this preconditioner in `Parameters`. How should this developer find all places in the code that need to be adjusted in case you do not add an Assert now.